Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -0,0 +1,1155 @@
+        private BoundsControlBoxDisplay boxDisplay = new BoundsControlBoxDisplay();
+        /// <summary>
+        /// Bounds control box display configuration section.
+        /// </summary>
+        public BoundsControlBoxDisplay BoxDisplay => boxDisplay;
+
+        [SerializeField]
+        [Tooltip("This section defines the links / lines that are drawn between the corners of the control.")]
+        private BoundsControlLinks links = new BoundsControlLinks();
+        /// <summary>
+        /// This section defines the links / lines that are drawn between the corners of the control.
+        /// </summary>
+        public BoundsControlLinks Links => links;
+
+        [SerializeField]
+        [Tooltip("Configuration of the scale handles.")]
+        private BoundsControlScaleHandles scaleHandles = new BoundsControlScaleHandles();
+        /// <summary>
+        /// Configuration of the scale handles.
+        /// </summary>
+        public BoundsControlScaleHandles ScaleHandles => scaleHandles;
+
+        [SerializeField]
+        [Tooltip("Configuration of the rotation handles.")]
+        private BoundsControlRotationHandles rotationHandles = new BoundsControlRotationHandles();
+        /// <summary>
+        /// Configuration of the rotation handles.
+        /// </summary>
+        public BoundsControlRotationHandles RotationHandles => rotationHandles;
+
+        [SerializeField]
+        [Tooltip("Configuration for Proximity Effect to scale handles or change materials on proximity.")]
+        private ProximityEffect proximityEffect = new ProximityEffect();
+        /// <summary>
+        /// Configuration for Proximity Effect to scale handles or change materials on proximity.
+        /// </summary>
+        public ProximityEffect ProximityEffect => proximityEffect;
+
+        [Header("Debug")]
+        [Tooltip("Debug only. Component used to display debug messages.")]
+        /// <summary>
+        /// Component used to display debug messages.
+        /// </summary>
+        public TextMesh debugText;
+
+        [SerializeField]
+        [Tooltip("Determines whether to hide GameObjects (i.e handles, links etc) created and managed by this component in the editor")]
+        private bool hideElementsInInspector = true;
+
+        /// <summary>
+        /// Determines whether to hide GameObjects (i.e handles, links etc) created and managed by this component in the editor
+        /// </summary>
+        public bool HideElementsInInspector
+        {
+            get { return hideElementsInInspector; }
+            set
+            {
+                if (hideElementsInInspector != value)
+                {
+                    hideElementsInInspector = value;
+                    UpdateRigVisibilityInInspector();
+                }
+            }
+        }
+
+        [Header("Events")]
+        /// todo: these should be somehow unified with manipulation handler and documented
```

Comment:
Are you planning to do that in this PR?

Reply:
no, theres's a couple of todos in there that i have to take a look at still - i just wanted to get rid of the first big junk for now because the PR kept growing. also wanted to get some feedback on it before i keep on going down that rabbit hole :)